sql/src/main/java/org/apache/druid/sql/calcite/planner/QueryUtils.java

```diff
@@ -55,4 +58,21 @@ public static List<ColumnMapping> buildColumnMappings(
 
     return columnMappings;
   }
+
+  public static JoinAlgorithm getJoinAlgorithm(Join join, PlannerContext plannerContext)
+  {
+    RelHint closestHint = null;
```

Comment:
note: I was thinking to use `DruidJoinHint` here.. so that you can't even set closesHint to a non-join hint...but its not really important

Reply:
Keeping the RelHint allows keeping track the closestHint's path length, to compare in the for loop. We do have a check to see if it's a join hint in the loop so this should be fine